ndnfs/daemon/driver.go

```diff
@@ -5,68 +5,233 @@ import (
 	log "github.com/Sirupsen/logrus"
 	"sync"
 	"github.com/docker/go-plugins-helpers/volume"
-	"github.com/Nexenta/nedge-docker-nfs/ndnfs/ndnfsapi"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"os/exec"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"strconv"
 )
 
+const defaultChunkSize int = 32768;
+const defaultMountPoint string = "/var/lib/ndnfs"
+
 var (
 	DN = "ndnfsdriver "
 )
 
 type NdnfsDriver struct {
 	Scope		string
 	DefaultVolSz	int64
-	Client		*ndnfsapi.Client
 	Mutex		*sync.Mutex
+	Endpoint    string
+	Config      *Config
 }
 
-func DriverAlloc(cfgFile string) NdnfsDriver {
+type Config struct {
+	Name        string
+	Nedgerest   string
+	Nedgeport   int16
+	Nedgedata	string
+	Clustername string
+	Tenantname  string
+	Chunksize   int
+	Username	string
+	Password	string
+	Mountpoint  string
+	Servicename string
+}
+
+func ReadParseConfig(fname string) (Config, error) {
+	content, err := ioutil.ReadFile(fname)
+	if err != nil {
+		log.Fatal(DN, "Error reading config file: ", fname, " error: ", err)
+	}
+	var conf Config
+	err = json.Unmarshal(content, &conf)
+	if err != nil {
+		log.Fatal(DN, "Error parsing config file: ", fname, " error: ", err)
+	}
+	return conf, nil
+}
 
-	client, _ := ndnfsapi.ClientAlloc(cfgFile)
+func DriverAlloc(cfgFile string) NdnfsDriver {
+	conf, err := ReadParseConfig(cfgFile)
```

Comment:
At this moment either 'err' equals nil or ReadParseConfig exists the application on error with 'log.Fatal'. Are you going to add returning errors from ReadParseConfig later?